### PR TITLE
fix: define local data types to each module

### PR DESCRIPTION
**Pull request by simensrostad**
- define local data types for data sent from the module
 - Make so that modules only transmit their own events, no mimic.
 - Split modem data into static and dynamic modem data.

### CODE REVIEW

File: src/events/gps_module_event.h [hunk 1]
```diff
@@ -33,13 +32,30 @@ enum gps_module_event_type {
 	GPS_EVT_ERROR_CODE,
 };
 
+struct gps_module_data {
+	/** GPS data timestamp. UNIX milliseconds. */
+	int64_t gps_ts;
```
Comment (jtguggedal) [inline src/events/gps_module_event.h]: `timestamp`?

File: src/events/gps_module_event.h [hunk 2]
```diff
@@ -33,13 +32,30 @@ enum gps_module_event_type {
...
+	/** Longitude */
+	double longi;
```
Comment (jtguggedal) [inline src/events/gps_module_event.h]: `Longitude`? Generally, I'd avoid abbreviations in these new structs. Then comments can also be removed when the struct member names are self-explanatory

File: src/modules/data_module.c [hunk 3]
```diff
@@ -696,54 +690,109 @@ static void on_all_states(struct data_msg_data *msg)
 	}
 
 	if (IS_EVENT(msg, ui, UI_EVT_BUTTON_DATA_READY)) {
-		cloud_codec_populate_ui_buffer(
-					ui_buf,
-					&msg->module.ui.data.ui,
-					&head_ui_buf);
+
```
Comment (jtguggedal) [inline src/modules/data_module.c]: Stray blank line, can be removed

File: src/modules/data_module.c [hunk 4]
```diff
@@ -696,54 +690,109 @@ static void on_all_states(struct data_msg_data *msg)
...
 		SEND_EVENT(data, DATA_EVT_UI_DATA_READY);
 		return;
 	}
 
 	if (IS_EVENT(msg, modem, MODEM_EVT_MODEM_DATA_READY)) {
-		cloud_codec_populate_modem_buffer(
-			modem_buf,
-			&msg->module.modem.data.modem,
-			&head_modem_buf);
+
```
Comment (jtguggedal) [inline src/modules/data_module.c]: Blank line not needed

File: src/modules/data_module.c [hunk 5]
```diff
@@ -696,54 +690,109 @@ static void on_all_states(struct data_msg_data *msg)
...
+						&head_bat_buf);
 
 		data_status_set(APP_DATA_BATTERY);
 	}
 
 	if (IS_EVENT(msg, sensor, SENSOR_EVT_ENVIRONMENTAL_DATA_READY)) {
-		cloud_codec_populate_sensor_buffer(
-			sensors_buf,
-			&msg->module.sensor.data.sensors,
-			&head_sensor_buf);
```
Comment (jtguggedal) [inline src/modules/data_module.c]: blank

File: src/modules/data_module.c [hunk 6]
```diff
@@ -696,54 +690,109 @@ static void on_all_states(struct data_msg_data *msg)
...
+	if (IS_EVENT(msg, sensor, SENSOR_EVT_ENVIRONMENTAL_NOT_SUPPORTED)) {
 		data_status_set(APP_DATA_ENVIRONMENTAL);
 	}
 
 	if (IS_EVENT(msg, sensor, SENSOR_EVT_MOVEMENT_DATA_READY)) {
-		cloud_codec_populate_accel_buffer(
-			accel_buf,
-			&msg->module.sensor.data.accel,
-			&head_accel_buf);
+
```
Comment (jtguggedal) [inline src/modules/data_module.c]: blank

File: src/modules/data_module.c [hunk 7]
```diff
@@ -696,54 +690,109 @@ static void on_all_states(struct data_msg_data *msg)
...
+		cloud_codec_populate_accel_buffer(accel_buf, &new_movement_data,
+						  &head_accel_buf);
 	}
 
 	if (IS_EVENT(msg, gps, GPS_EVT_DATA_READY)) {
-		cloud_codec_populate_gps_buffer(
-			gps_buf,
-			&msg->module.gps.data.gps,
-			&head_gps_buf);
+
```
Comment (jtguggedal) [inline src/modules/data_module.c]: blank

**Review by jhn-nordic — APPROVED** [review]

File: src/modules/modem_module.c [hunk 8]
```diff
@@ -254,43 +254,94 @@ static int modem_data_get(void)
...
+		LOG_ERR("modem_info_params_get, error: %d", err);
+		return err;
+	}
+
+	struct modem_module_event *modem_module_event =
+			new_modem_module_event();
+
+	modem_module_event->data.modem_dynamic.rsrp =
+			rsrp_value_latest;
+
```
Comment (jtguggedal) [inline src/modules/modem_module.c:306]: The blank lines between variable assignments are not really needed

**Review by jtguggedal — APPROVED** [review]

File: src/modules/modem_module.c (hunk 8 above)
Comment (simensrostad) [reply src/modules/modem_module.c:306]: woops. I merged before I saw your comment.

**Pull request merged**